prefect_gcp/cloud_storage.py

```diff
@@ -1140,3 +1145,77 @@ async def upload_from_folder(
         if num_uploaded == 0:
             self.logger.warning(f"No files were uploaded from {from_folder}.")
         return bucket_folder
+
+    @sync_compatible
+    async def upload_from_dataframe(
+        self,
+        df: DataFrame,
+        to_path: str,
+        output_format: Literal["parquet", "csv"] = "csv",
+        compression: Optional[str] = None,
+        **upload_kwargs: Dict[str, Any],
+    ) -> str:
+        """
+        Upload a pandas Dataframe to Google Cloud Storage as:
+            .csv, .csv.gz, .parquet, .parquet.snappy, .parquet.gz
+
+        Args:
+            df: The pandas Dataframe object to upload.
+            to_path: The path to upload the object to; this gets prefixed
+                with the bucket_folder.
+            output_format: The output format to serialize the dataframe into.
+            compression: Specify the compression type for the output.
+                'csv' supports 'None' or 'gzip',
+                'parquet' supports 'None', 'snappy', 'gzip'.
+            **upload_kwargs: Additional keyword arguments to pass to
+                `Blob.upload_from_dataframe`.
+
+        Returns:
+            The path that the folder was uploaded to.
+
+        Examples:
+            Upload local folder my_folder to the bucket's folder my_folder.
+            ```python
+            from prefect_gcp.cloud_storage import GcsBucket
+
+            gcs_bucket = GcsBucket.load("my-bucket")
+            gcs_bucket.upload_from_dataframe(
+                df=pandas_df, to_path="/path/to/gcs/blob.parquet",
+                output='parquet', compression='snappy'
+            )
+            ```
+        """
+        if output_format not in {"csv", "parquet"}:
+            raise ValueError("Unsupported output format. Use either 'csv' or 'parquet'")
+
+        if output_format == "csv" and compression not in {None, "gzip"}:
+            raise ValueError(
+                "Unsupported compression format for .csv. " "Use either 'gzip' or None"
+            )
+        if output_format == "parquet" and compression not in {None, "gzip", "snappy"}:
+            raise ValueError(
+                "Unsupported compression format for .parquet. "
+                "Use either 'None', 'gzip', or 'snappy'"
+            )
+
+        content_type: str = "application/octet-stream"
+
+        with BytesIO() as bytes_buffer:
+            if output_format == "csv":
+                df.to_csv(
+                    path_or_buf=bytes_buffer, compression=compression, index=False
+                )
+                if compression is None:
+                    content_type = "text/csv"
+
+            else:
+                df.to_parquet(path=bytes_buffer, compression=compression, index=False)
+
+            bytes_buffer.seek(0)
+            gcs_path = await self.upload_from_file_object(
+                from_file_object=bytes_buffer,
+                to_path=to_path,
```

Comment:
To be consistent with other methods.
```suggestion
            bucket_path = self._join_bucket_folder(to_path)
            gcs_path = await self.upload_from_file_object(
                from_file_object=bytes_buffer,
                to_path=bucket_path,
```

Reply:
@ahuang11 

I can't really do `bucket_path = self._join_bucket_folder(to_path)`
At the end, my method calls `self.upload_from_file_object`, which already has a `self._join_bucket_folder(to_path)`


It seems that if we call self._join_bucket_folder multiple times, it appending the `base_folder` to the prefix

<img width="942" alt="image" src="https://user-images.githubusercontent.com/901651/218279015-d83c922a-15c1-4892-833d-5b99c9cf0dc6.png">

I'm not sure that's the expected behavior  on that `_join_bucket_folder` method, but that's how it is set, right now.

Reply:
@ahuang11 Any thoughts on this?

Reply:
Apologies for late response, I was out of office, and you're right; if we call it again, it'll double the prefix.